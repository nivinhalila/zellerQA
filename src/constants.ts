import { ZellerAdmin, ZellerManager } from "./types";

export const ZELLER_ADMIN: ZellerAdmin = "ADMIN";
export const ZELLER_MANAGER: ZellerManager = "MANAGER";
