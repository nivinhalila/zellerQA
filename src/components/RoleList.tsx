import React from "react";

import RoleListItem from "./RoleListItem";
import Section from "./Section";
import Title from "./Title";

import { ZellerRoles } from "../types";
import { ZELLER_ADMIN, ZELLER_MANAGER } from "../constants";

interface RoleListProps {
  role: string;
  setRole: (role: ZellerRoles) => void;
}

export function RoleList(props: RoleListProps) {
  const { role, setRole } = props;

  return (
    <Section>
      <Title>User Types</Title>
      <RoleListItem
        handleOnChange={() => setRole(ZELLER_ADMIN)}
        isSelected={role === ZELLER_ADMIN}
        label="Admin"
      />
      <RoleListItem
        handleOnChange={() => setRole(ZELLER_MANAGER)}
        isSelected={role === ZELLER_MANAGER}
        label="Manager"
      />
    </Section>
  );
}

export default RoleList;
