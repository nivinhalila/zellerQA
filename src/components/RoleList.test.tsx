import "jest-styled-components";
import { render, fireEvent } from "@testing-library/react";

import RoleList from "./RoleList";
import { ZELLER_ADMIN, ZELLER_MANAGER } from "../constants";

test("renders role list component", () => {
  expect.assertions(4);

  const role = ZELLER_ADMIN;
  const setRole = jest.fn();

  const { container, getByText } = render(
    <RoleList role={role} setRole={setRole} />
  );

  expect(getByText("User Types")).toBeInTheDocument();
  expect(getByText("Admin")).toBeInTheDocument();
  expect(getByText("Manager")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});

test("calls set role when admin label clicked", () => {
  expect.assertions(2);

  const role = ZELLER_MANAGER;
  const setRole = jest.fn();

  const { getByText } = render(<RoleList role={role} setRole={setRole} />);

  fireEvent.click(getByText(/Admin/));

  expect(setRole).toBeCalledTimes(1);
  expect(setRole).toBeCalledWith(ZELLER_ADMIN);
});

test("calls set role when manager label clicked", () => {
  expect.assertions(2);

  const role = ZELLER_ADMIN;
  const setRole = jest.fn();

  const { getByText } = render(<RoleList role={role} setRole={setRole} />);

  fireEvent.click(getByText(/Manager/));

  expect(setRole).toBeCalledTimes(1);
  expect(setRole).toBeCalledWith(ZELLER_MANAGER);
});
