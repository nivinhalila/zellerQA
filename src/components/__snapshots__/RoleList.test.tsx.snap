// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders role list component 1`] = `
.c4 {
  color: #212322;
  font-size: 1.5em;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #e8f2fb;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 1rem;
  padding: 0rem 2rem;
}

.c2:last-child {
  margin-bottom: 0;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 1rem;
  padding: 0rem 2rem;
}

.c5:last-child {
  margin-bottom: 0;
}

.c3 {
  height: 2rem;
  margin: 0 1.5rem 0 0;
  width: 2rem;
}

.c0 {
  background: #ffffff;
  margin-bottom: 2.5rem;
}

.c1 {
  color: #212322;
  font-size: 2em;
  font-weight: bold;
}

<section
  class="c0"
>
  <h1
    class="c1"
  >
    User Types
  </h1>
  <label
    class="c2"
    for="Admin"
  >
    <input
      checked=""
      class="c3"
      data-testid="Admin"
      id="Admin"
      name="role"
      type="radio"
    />
    <p
      class="c4"
    >
      Admin
    </p>
  </label>
  <label
    class="c5"
    for="Manager"
  >
    <input
      class="c3"
      data-testid="Manager"
      id="Manager"
      name="role"
      type="radio"
    />
    <p
      class="c4"
    >
      Manager
    </p>
  </label>
</section>
`;
